nav2_safety_nodes/src/nav2_safety_node.cpp

```diff
@@ -0,0 +1,250 @@
+  declare_parameter("tf_tolerance", rclcpp::ParameterValue(0.01));
+  declare_parameter("scan_topics", scan);
+  declare_parameter("speed_limit_topic", rclcpp::ParameterValue("speed_limit"));
+}
+
+SafetyZone::~SafetyZone()
+{
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Configuring");
+  getParameters();
+  initTransforms();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Activating");
+  initPubSub();
+  safety_polygon_pub_->on_activate();
+  point_cloud_pub_->on_activate();
+  speed_limit_pub_->on_activate();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Deactivating");
+  safety_polygon_pub_->on_deactivate();
+  point_cloud_pub_->on_deactivate();
+  speed_limit_pub_->on_deactivate();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_cleanup(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Cleaning up");
+  safety_polygon_pub_.reset();
+  point_cloud_pub_.reset();
+  speed_limit_pub_.reset();
+  timer_.reset();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_shutdown(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(logger_, "Shutting down");
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+// Get Parameters
+void
+SafetyZone::getParameters()
+{
+  RCLCPP_DEBUG(logger_, " getParameters");
+  std::string speed_limit_topic;
+  // Get all of the required parameters
+  safety_polygon_ = get_parameter("safety_polygon").as_string();
+  zone_action_ = get_parameter("zone_action").as_double();
+  zone_priority_ = get_parameter("zone_priority").as_int();
+  zone_num_pts_ = get_parameter("zone_num_pts").as_int();
+  base_frame_ = get_parameter("base_frame").as_string();
+  tf_tolerance_ = get_parameter("tf_tolerance").as_double();
+  scan_topics_ = get_parameter("scan_topics").as_string_array();
+  speed_limit_topic_ = get_parameter("speed_limit_topic").as_string();
+
+  // If the safety_polygon has been specified, it must be in the correct format
+  if (safety_polygon_ != "" && safety_polygon_ != "[]") {
+    // Polygon parameter has been specified, polygon -> point vector(safety_zone)
+    std::vector<geometry_msgs::msg::Point> safety_zone;
+    /// string of polygon points and returns a polygon vector
+    nav2_util::makeVectorPointsFromString(safety_polygon_, safety_zone);
+  } else {
+    // Polygon provided but invalid, so stay with the radius
+    RCLCPP_ERROR(
+      logger_, "The safety_polygon is invalid: \"%s\" :) ",
+      safety_polygon_.c_str());
+  }
+}
+
+void
+SafetyZone::initTransforms()
+{
+  RCLCPP_INFO(get_logger(), "initTransforms");
```

Comment:
Delete this function and move its contents into on_configure

On logging, same with configure, just have a log for in on configure, and I'm setting up. Something simple.